js/lightbeam.js

```diff
@@ -1,9 +1,52 @@
+// transforms the object of nested objects 'websites' into a
+// usable format for d3
+function transformData(websites) {
+  const nodes = [];
+  const links = [];
+  const firstParties = new Set();
+  const thirdParties = new Set();
+  // make first party nodes
+  for (const firstParty in websites) {
+    const firstPartyNode = createWebsite(firstParty);
+    firstParties.add(firstPartyNode);
+    nodes.push(firstPartyNode);
+    // make third party nodes and links
+    if (websites[firstParty].thirdPartyRequests) {
+      for (const thirdParty in websites[firstParty].thirdPartyRequests) {
+        const thirdPartyNode = createWebsite(thirdParty);
+        thirdParties.add(thirdPartyNode);
+        if (!firstParties.has(thirdPartyNode)) {
```

Comment:
This won't work, I thought we were just going to have a set of hostnames. Either way lets do this as a follow up which is fine.